lib/winapi/timezoneapi.c

```diff
@@ -0,0 +1,205 @@
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_STD_DATE, &type, &daylightEndDate, sizeof(daylightEndDate), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_DLT_BIAS, &type, &daylightBias, sizeof(daylightBias), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_STD_BIAS, &type, &standardBias, sizeof(standardBias), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_DLT_NAME, &type, daylightName, sizeof(daylightName), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_STD_NAME, &type, standardName, sizeof(standardName), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+
+    if ((XboxTimeZoneValid(&daylightStartDate) == FALSE) || (XboxTimeZoneValid(&daylightEndDate) == FALSE)) {
+        return TIME_ZONE_ID_INVALID;
+    }
+
+    // Xbox stores up to 4 characters here
+    for (UCHAR i = 0; i < 4; i++) {
+        lpTimeZoneInformation->StandardName[i] = standardName[i];
+        lpTimeZoneInformation->DaylightName[i] = daylightName[i];
+    }
+
+    // There was no DST info
+    if (daylightEndDate.Month == 0 || daylightStartDate.Month == 0) {
+        return TIME_ZONE_ID_UNKNOWN;
+    }
+
+    XboxTimeZoneToSystemTime(&lpTimeZoneInformation->DaylightDate, &daylightStartDate);
+    XboxTimeZoneToSystemTime(&lpTimeZoneInformation->StandardDate, &daylightEndDate);
+    lpTimeZoneInformation->DaylightBias = daylightBias;
+    lpTimeZoneInformation->StandardBias = standardBias;
+
+    // Now we need to determine if the current time is within the DST range
+    KeQuerySystemTime(&kTime);
+    RtlTimeToTimeFields(&kTime, &dateNow);
+
+    // Determine what day in the month DST starts and ends
+    UCHAR daylightStartDay = GetActualDay(dateNow.Year, &daylightStartDate);
+    UCHAR daylightEndDay = GetActualDay(dateNow.Year, &daylightEndDate);
+
+    // Fix month wrapping. i.e if DST is from Oct(10) to Feb(2) and the current month is Jan(1) we should be in DST
+    // Applying a simple check would result is 10 < 1 < 2 = false and incorrect
+    // To account for this, we offset the months in the following year by 12
+    // This results in the comparison 10 < 13(12+1) < 14(12+2) = true and is correct
+    if (daylightEndDate.Month < daylightStartDate.Month)
+    {
```

Comment:
nit: Inconsistent wether `{` is on new line

Reply:
done